stream_alert/rule_processor/rules_engine.py

```diff
@@ -403,3 +426,38 @@ def process(cls, input_payload):
                     alerts.append(alert)
 
         return alerts
+
+    @classmethod
+    def load_intelligence(cls, intel_dir='threat_intel'):
+        """Load intelligence from csv.gz files into a dictionary
+
+        Args:
+            intel_dir (str): Location where stores compressed intelligence
+        """
+        if cls.__intelligence:
+            return
+        threat_intel = StreamThreatIntel(intel_dir)
+        cls.__intelligence = threat_intel.read_compressed_files()
+
+    @classmethod
+    def is_ioc(cls, rec, datatypes_ioc_mapping=None):
+        """Detect if a record contains any data matching to known IOCs
+
+        A class variable is defined to store IOCs (Indicator of Compromise) in a
+        dictionary.
+
+        Args:
+            rec (dict): Payload record to process
+
+        Return:
+            (tuple): (True or False, IOC type, IOC value)
+        """
+        if datatypes_ioc_mapping:
```

Comment:
suggestion here to avoid multi-level nesting so much would be to return early if possible. ie:
```
if not datatypes_ioc_mapping:
    return False, None, None
```